src/server/core.rs

```diff
@@ -37,6 +38,13 @@ pub struct Core {
     secret_key: Option<Arc<SecretKey>>,
 
     custom_conn_handler: Option<CustomConnectionHandler>,
+    event_queue: Vec<WorkerEvent<FromWorkerMessage>>,
+}
+
+pub struct WorkerEvent<T> {
```

Comment:
Rename to `MonitoringEvent` and remove generics, it will contain an enum (`MonitoringEventPayload` or some name like this). Let's move these events to a separate file, they are not related to gateway.